### ISSUE TITLE
authentication-oauth triggers express deprecation warning

### ISSUE THREAD

**Issue by raphaelschwinger** [feathersjs/feathers#2246]
Hey first of all thanks for your great work!

I think I run into a bug that is caused by accessing the [deprecated](https://github.com/feathersjs/feathers/blob/crow/.github/contributing.md#report-a-bug) `router` function of express.

### Steps to reproduce
* install feathers `5.0.0-pre.1`:
* follow auth [guide](https://docs.feathersjs.com/guides/basics/authentication.html#github-login-oauth)
*  build project (we use esbuild)

### Expected behavior
Feathers starts as expected and oauth with github works

### Actual behavior
App throws an error at: `pnp:/workspace/.yarn/cache/express-npm-4.17.1-6815ee6bf9-c4b470d623.zip/node_modules/express/lib/application.js`
```
 throw new Error("'app.router' is deprecated!\nPlease see the 3.x to 4.x migration guide for details on how to update your app.");
```
called by: '@feathersjs/feathers/lib/hooks/index.js'
```
service.methods = Object.getOwnPropertyNames(service).filter((key) => typeof service[key] === "function" && service[key][ACTIVATE_HOOKS]).reduce((result, methodName) => {
      result[methodName] = service[methodName][ACTIVATE_HOOKS];
      return result;
    }, service.methods || {});
```
### System configuration
* dockerized `node:lts`

**Module versions** (especially the part that's not working):
```
    "@feathersjs/authentication": "5.0.0-pre.1",
    "@feathersjs/authentication-oauth": "5.0.0-pre.1",
    "@feathersjs/express": "5.0.0-pre.1",
    "@feathersjs/feathers": "5.0.0-pre.1",
    "@feathersjs/socketio": "5.0.0-pre.1"
```

**NodeJS version**:
`v14.15.5`

**Comment by daffl** [comment]
Can you elaborate when this is happening? I have not seen this error before and `app.router` is not used anywhere in the Feathers codebase.

**Comment by raphaelschwinger** [comment]
Hey @daffl as far as I understand it is  the `router` "function" added to the `authApp` when it gets initialized by calling `express()`. Its only purpose is to warn that a deprecated  function is called. When iterating through all properties of the `authApp` at '@feathersjs/feathers/lib/hooks/index.js':
```
service.methods = Object.getOwnPropertyNames(service).filter((key) => typeof service[key] === "function" && service[key][ACTIVATE_HOOKS]).reduce((result, methodName) => {
      result[methodName] = service[methodName][ACTIVATE_HOOKS];
      return result;
    }, service.methods || {});
```
this warning gets triggers and causes mentioned runtime error.

**Comment by daffl** [comment]
Ah yes, apologies, I also realized you were talking about the prerelease. The code in question has been refactored so I believe this should be fixed in the `v5.0.0-beta.0` I just released.

**Comment by daffl** [comment]
Just tested it with oAuth and it seems to be working so closing this issue. Feel free to reopen if it is still around in the latest prerelease.
